x-pack/test/apm_api_integration/tests/traces/generate_trace.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { apm } from '@kbn/apm-synthtrace-client';
+
+type Instance = ReturnType<ReturnType<typeof apm.service>['instance']>;
+type Transaction = ReturnType<Instance['transaction']>;
+
+export function generateTrace(
+  timestamp: number,
+  order: Instance[],
+  db?: 'elasticsearch' | 'redis'
+) {
+  return order
+    .concat()
+    .reverse()
+    .reduce<Transaction | undefined>((prev, instance, index) => {
+      const invertedIndex = order.length - index - 1;
+
+      const duration = 50;
+      const time = timestamp + invertedIndex * 10;
+
+      const transaction: Transaction = instance
+        .transaction({ transactionName: `GET /${instance.fields['service.name']!}/api` })
+        .timestamp(time)
+        .duration(duration);
+
+      if (prev) {
+        const next = order[invertedIndex + 1].fields['service.name']!;
+        transaction.children(
+          instance
+            .span({ spanName: `GET ${next}/api`, spanType: 'external', spanSubtype: 'http' })
+            .destination(next)
+            .duration(duration)
+            .timestamp(time + 1)
+            .children(prev)
+        );
+      } else if (db) {
+        transaction.children(
+          instance
+            .span({ spanName: db, spanType: 'db', spanSubtype: db })
+            .destination(db)
+            .duration(duration)
+            .timestamp(time + 1)
+        );
+      }
```

Comment:
I created some shorthands that creates the service destination spans:
https://github.com/elastic/kibana/blob/f2b8c7f90795fad742dd626ab818a9a4795f4aff/packages/kbn-apm-synthtrace-client/src/lib/dsl/distributed_trace_client.test.ts#L40

They might be useful to you. Either by using `DistributedTrace` class, or by extracting some of the methods to use separately.